js/src/rive_advanced.mjs.d.ts

```diff
@@ -1,325 +1,669 @@
+   * when no longer in use
+   */
+  delete(): void;
+}
+
+/**
+ * Rive class representing a StateMachine instance. Use this class to advance and control a
+ * particular state machine in the render loop (i.e scrub, grab state machine inputs, set up
+ * listener events, etc.).
+ *
+ * Important: Make sure to delete this instance when it's no longer in use via the `delete()`
+ * method. This deletes the underlying c++ reference and frees up the backing WASM object. This can
+ * be done in cases where the user navigates away from the page with this animation, the canvas is
+ * unmounted, etc.
+ */
+export declare class StateMachineInstance {
+  /**
+   * Create a new StateMachineInstance reference
+   * @param stateMachine - A StateMachineInstance retrieved via the Artboard
+   * (i.e `artboard.stateMachineByName('foo')`)
+   * @param artboard - The Artboard instance for this state machine
+   */
+  constructor(stateMachine: StateMachineInstance, artboard: Artboard);
+  get name(): string;
+  /**
+   * Returns the number of inputs associated with this state machine
+   * @returns Number of inputs
+   */
+  inputCount(): number;
+  /**
+   * Returns the state machine input at the given index
+   * @param i - Index to retrieve the state machine input at
+   * @returns SMIInput reference
+   */
+  input(i: number): SMIInput;
+  /**
+   * Advances/scrubs the StateMachineInstance by the set amount of seconds. Note that this does not
+   * apply changes to the properties of objects in the Artboard yet.
+   * @param sec - Scrub the state machine instance by a number of seconds
+   */
+  advance(sec: number): boolean;
+  /**
+   * Returns the number of states changed while the state machine played
+   * @returns Number of states changed in the duration of the state machine played
+   */
+  stateChangedCount(): number;
+  /**
+   * Returns the name of the state/animation transitioned to, given the index in the array of state
+   * changes in total
+   * @param i
+   * @returns Name of the state/animation transitioned to
+   */
+  stateChangedNameByIndex(i: number): string;
+
+  /**
+   * Notifies the state machine that the pointer has pressed down at the given coordinate in
+   * Artboard space. Internally, Rive may advance a state machine if the listener coordinate is of
+   * interest at a given moment.
+   * 
+   * @param x - X coordinate
+   * @param y - Y coordinate
+   */
+  pointerDown(x: number, y: number): void;
+  /**
+   * Notifies the state machine that the pointer has pressed down at the given coordinate in
+   * Artboard space. Internally, Rive may advance a state machine if the listener coordinate is of
+   * interest at a given moment.
+   * 
+   * @param x - X coordinate
+   * @param y - Y coordinate
+   */
+  pointerMove(x: number, y: number): void;
+  /**
+   * Registers a coordinate where the pointer up listener watches for in the Artboard space.
```

Comment:
Notifies the state machine that the pointer has released at ...

Reply:
bleh - good catch on these!